data/statements/cms-contest.cls

```diff
@@ -401,6 +403,38 @@
 }
 
 
+% Tools to extract subtask and task limits and scores.
+\newwrite\py@num@tmp
+\newcommand{\pynum}[2][]{%
+\immediate\openout\py@num@tmp=\currfiledir/py_num.py%
+\if\relax\detokenize{#1}\relax
+\else
+\immediate\write\py@num@tmp{from #1 import *}%
+\fi
+\immediate\write\py@num@tmp{print(#2)}%
+\immediate\closeout\py@num@tmp%
+\immediate\write18{cd \currfiledir; python py_num.py > ../py_num.tex}%
+\CatchFileDef{\py@num}{py_num.tex}{}%
+\num\py@num%
+}
+
+\newcommand{\limiti}[1]{%
```

Comment:
This should be updated according to the discussion we had in the Telegram group. If I understood correctly it should be something like: 

- `\limiti{<stmt>}` → `from limiti import *; print(<stmt>)`
- `\limiti[<n>]{<name>}` → `from limiti import *; print(subtasks[<n>]["<name>"])`
- `\limitipy{<stmt>}` → `from limiti import *; print(<stmt>)`
- `\limitipy[<file>]{<stmt>}` → `from <file> import *; print(<stmt>)`

Now that I write it, it doesn't seem right... Especially since `\limitipy` can read from any file other than `limiti.py` 🤦🏻 

A better solution may be:

- `\pynum` which is exactly like you wrote (`\pynum[<file>]{<expr>}`)
- `\limiti{<expr>}` → `from limiti import *; print(<stmt>)`
- `\limiti[<n>]{<name>}` → `from limiti import *; print(subtasks[<n>]["<name>"])`

Even though this way `\limiti` does 2 very different things (evaluate in `limiti.py`, and read from a variable with a very specific format) just depending on an optional argument. Probably a nicer API would be if the third version is a different macro that takes 2 required params. Here some name suggestions: `\limitist{st}{name}`, `\subtask{st}{name}`, `\subtasklim{st}{name}`. The one I like the most is the first one.

cc: @harniver

Reply:
See newest version: \limitist{name}, or \limitist[2]{name} for overrides.